plugins/outputs/graphite/graphite.go

```diff
@@ -135,10 +176,13 @@ func (g *Graphite) Write(metrics []telegraf.Metric) error {
 
 	err = g.send(batch)
 
-	// try to reconnect and retry to send
-	if err != nil {
-		g.Log.Error("Graphite: Reconnecting and retrying...")
-		_ = g.Connect()
+	// If a send failed for a server, try to reconnect to that server
+	if len(g.failedServers) > 0 {
+		g.Log.Debugf("Graphite: Reconnecting and retrying for the following servers: %s", strings.Join(g.failedServers, ","))
```

Comment:
I realize the previous message had graphite in it, but no need since it will be prefaced by log messages with `inputs.graphite`

```suggestion
		g.Log.Debugf("Reconnecting and retrying for the following servers: %s", strings.Join(g.failedServers, ","))
```